### PR TITLE
Fluvio publish in release workflow

### LINKED ISSUE #465: Integrate package publisher with GitHub Release workflow
Release workflow uses package publisher int the GitHub Actions to publish to packages S3 bucket and github releases.
- CLI

Ideally, this will be done using fluvio-packages for the S3 bucket bit.

### PR DESCRIPTION
**Pull request by simlay**
I forgot that releases workflows can only run from a branch in the same repo.

This is the same PR as #530 and will close #465.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -139,13 +139,15 @@ check-clippy:	install-clippy
...
 
-build-all-test:	
+build-all-test:
 	cargo build --tests --all-features
 
 run-all-unit-test:
 	cargo test --all-features
 
 install_musl:
+	sudo apt-get update;
```
Comment (sehz) [inline Makefile:149]: this need to be specifically target toward ubuntu which make it not suitable here

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
+        os: [infinyon-ubuntu-bionic, macos-latest]
```
Comment (sehz) [inline .github/workflows/release.yml]: we should stick with GitHub runner for now

File: makefiles/Base.toml [hunk 3]
```diff
@@ -8,4 +8,39 @@ CARGO_PROFILE = "release"
...
+    "build",
+    "--release",
+    "--bin", "fluvio",
+    "--target=x86_64-unknown-linux-musl",
+    "@@split(CARGO_MAKE_CARGO_BUILD_TEST_FLAGS, )"
+]
+
+[tasks.setup-musl.linux]
+script = '''
+which musl-gcc 2> /dev/null > /dev/null || {
```
Comment (sehz) [inline makefiles/Base.toml:34]: this is highly specific to ubuntu or OS that has `apt` and doesn't belong here.  it should be done by whatever infrastructure provisioning tools

**Review by sehz — CHANGES_REQUESTED** [review]

File: makefiles/Base.toml (hunk 3 above)
Comment (simlay) [reply makefiles/Base.toml:34]: Yeah, I've got mixed feelings about it. This is why the check for `musl-gcc` exists so it doesn't need to be installed. This step is also contingent on it being done with the "production" profile.

Comment (sehz) [reply makefiles/Base.toml:34]: but that creates dependency to specific infrastructure.  they should move to CI configuration or provisioning tools

File: makefiles/S3.toml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+]
+
+[tasks.publish-fluvio]
+dependencies = [
+    "build-release",
+]
+script = '''
+mkdir -p ~/.fluvio/extensions
+./target/${TARGET}/release/fluvio install fluvio/fluvio-package;
+./target/${TARGET}/release/fluvio package publish ./target/${TARGET}/release/fluvio --version=${FLUVIO_VERSION} --force;
```
Comment (simlay) [inline makefiles/S3.toml:41]: remove `--force.`

Comment (simlay) [reply makefiles/S3.toml]: [Build worked](https://github.com/infinyon/fluvio/actions/runs/397617509]. The only difference between https://github.com/infinyon/fluvio/commit/0e1e509727933328b29aace642e2257420054e25 and https://github.com/infinyon/fluvio/commit/4a6d916ba29acf2e8990cba1b79732b1b7e4bf0c is the `--force` here.

**Review by sehz — APPROVED** [review]

**Pull request merged**